src/Conversion/KrnlToAffine/ConvertKrnlToAffine.cpp

```diff
@@ -778,6 +782,23 @@ static LogicalResult interpretOperation(Operation *op, OpBuilder &builder,
       parallelLoop.getRegion().takeBody(loopToParallel.getRegion());
       Operation *yieldOp = &parallelLoop.getBody()->back();
       yieldOp->setOperands(reducedValues);
+      if (needParallelClause) {
+        // Use clause only for the first one (expected the outermost one).
+        // Ideally, we would generate here a single, multi-dimensional
+        // AffineParallelOp, and we would not need to reset the flag.
+        needParallelClause = false;
```

Comment:
Is this condition used afterwards?

Reply:
Yes, when we need the parallel clause, then only the first loop in the `for(Value loopRef : loopRefs)` will execute the addition of the `KrnlParallelClauseOp`